File: tests/rspec/pages/publish/publish.rb
class Publish < ControlPanelPage
  set_url '/admin.php?/cp/publish/create/{channel_id}'

  element :title, 'input[name=title]'
  element :url_title, 'input[name=url_title]'
  element :save, '.form-btns-top button[name=submit][value=save]'
  element :save_and_close, '.form-btns-top button[name=submit][value=save_and_close]'

  elements :file_fields, 'div[data-file-field-react]'
  elements :chosen_files, '.fields-upload-chosen img'

  elements :tab_links, 'ul.tabs li'
  elements :tabs, '.tab-wrap div.tabs'

  section :file_modal, FileModal, '.modal-file'
  section :forum_tab, ForumTab, 'body'
  section :fluid_field, FluidField, '.fluid-wrap'
end

File: tests/cypress/cypress/elements/pages/publish/Publish.js
import ControlPanel from '../ControlPanel'

class Publish extends ControlPanel {
  constructor() {
      super()
      this.url = '/admin.php?/cp/publish/create/{channel_id}';

      this.elements({
        'title': 'input[type!=hidden][name=title]',
        'url_title': 'input[type!=hidden][name=url_title]',
        'save': '.tab-bar__right-buttons .form-btns button[name=submit][value=save]',
        'save_and_close': '.tab-bar__right-buttons .form-btns button[name=submit][value=save_and_close]',

        'file_fields': 'div[data-file-field-react]',
        'chosen_files': '.fields-upload-chosen:visible img',

        'tab_links': '.tab-wrap div.tab-bar__tabs .tab-bar__tab',
        'tabs': '.tab-wrap div.tab-bar__tabs'

        // section :file_modal, FileModal, '.modal-file'
        // section :forum_tab, ForumTab, 'body'
        // section :fluid_field, FluidField, '.fluid-wrap'
      })
    }

}
export default Publish;